src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_FilterModel.cs

```diff
@@ -371,8 +371,11 @@ private bool IsAfterDot(Model model, ITextSnapshot textSnapshot, Dictionary<Text
 
                 if (bestFilterResult != null)
                 {
+                    // Only hard select this result if it's a prefix match
+                    var prefixLength = bestFilterResult.Value.CompletionItem.FilterText.GetCaseInsensitivePrefixLength(model.FilterText);
```

Comment:
isn't this just asking if the compiler item's FilterText 'StartsWith' the model's filtertext?